classes/local/factor/object_factor_base.php

```diff
@@ -77,12 +77,22 @@ public function add_user_factor($data) {
         return false;
     }
 
-    public function get_user_factors($user) {
+    public function get_all_user_factors($user) {
+        return array();
+    }
+
+    public function get_enabled_user_factor($userid) {
+        $factors = $this->get_all_user_factors($userid);
+        foreach ($factors as $factor) {
+            if ($factor->disabled == 0) {
+                return $factor;
```

Comment:
this doesn't look right, are you filtering this array?

Reply:
Corrected